### PR TITLE
Enable 'SHOW TOPICS' to work with secure kafka clusters

### LINKED ISSUE #740: SHOW TOPICS does not work against a secure Kafka cluster
Right now, if we use `show topics;` against a secure Kafka cluster, it will fail with something like:

```
java.lang.RuntimeException: Request METADATA failed on brokers List(r0.kafka-mt-1.us-west-2.aws.stag.cpdev.cloud:9092 (id: -1 rack: null), r0.kafka-mt-1.us-west-2.aws.stag.cpdev.cloud:9094 (id: -3 rack: null), r0.kafka-mt-1.us-west-2.aws.stag.cpdev.cloud:9093 (id: -2 rack: null))
```

The reason is that the [KafkaConsumerGroupClient](https://github.com/confluentinc/ksql/blob/master/ksql-engine/src/main/java/io/confluent/ksql/util/KafkaConsumerGroupClientImpl.java) does not pass along the security configs to the underlying kafka client when instantiated. 

```scala
 public List<String> listGroups() {
    Map<String, Object> clientConfigProps = ksqlConfig.getKsqlAdminClientConfigProps();
    String[] args = {
        "--bootstrap-server", (String) clientConfigProps.get("bootstrap.servers")
    };

    ConsumerGroupCommand.ConsumerGroupCommandOptions opts =
        new ConsumerGroupCommand.ConsumerGroupCommandOptions(args);
   ...
}
```

We should update the code so that all the client configs are passed along.

### PR DESCRIPTION
**Pull request by apurvam**
Fixes #740 

The `ConsumerGroupCommand` which we use to list groups instantiates its own admin client instance. The only way to configure this instance is to pass the configurations through a file.

This patch creates a temporary file, dumps the admin client properties to it, and then passes this file as an argument to the consumer group command.

I tested that this actually works against a secure cluster. Not sure what tests to add as part of this patch. Unit tests would require extensive mocking, which seems to be low ROI given that the use of the `ConsumerGroupCommand` itself is on the deprecation path. See #642 .

I am open to creative suggestions for low cost testing of this change.

### CODE REVIEW

File: ksql-engine/src/main/java/io/confluent/ksql/util/KafkaConsumerGroupClientImpl.java [hunk 1]
```diff
@@ -49,9 +57,22 @@ public KafkaConsumerGroupClientImpl(KsqlConfig ksqlConfig) {
 
   @Override
   public List<String> listGroups() {
+    File tmpConfigFile;
```
Comment (big-andy-coates) [inline ksql-engine/src/main/java/io/confluent/ksql/util/KafkaConsumerGroupClientImpl.java]: Consider moving all the parts needed to create the admin client args into a `getAdminClientArgs()`

File: ksql-engine/src/main/java/io/confluent/ksql/util/KafkaConsumerGroupClientImpl.java [hunk 2]
```diff
@@ -49,9 +57,22 @@ public KafkaConsumerGroupClientImpl(KsqlConfig ksqlConfig) {
...
+    try {
+      // The ConsumerGroupCommand we use instantiates its own admin client. However, the configs
+      // for the underlying admin client can be passed only through a properties file. So we dump
+      // the admin client configs to a temporary file and then use that file to configure the
+      // underlying admin client correctly.
+      tmpConfigFile = flushPropertiesToTempFile();
+    } catch (IOException e) {
+      log.error("Could not list consumer groups", e);
+      return Collections.emptyList();
```
Comment (big-andy-coates) [inline ksql-engine/src/main/java/io/confluent/ksql/util/KafkaConsumerGroupClientImpl.java]: Do we really want to swallow this exception? The `listGroups` can potentially throw, and that isn't being swallowed. It'll be a poor user experience if `SHOW TOPICS` returns an empty list to the user.

How about wrapping entire function in try/catch throwing an appropriate exception up the stack?

File: ksql-engine/src/main/java/io/confluent/ksql/util/KafkaConsumerGroupClientImpl.java [hunk 3]
```diff
@@ -67,6 +88,20 @@ public KafkaConsumerGroupClientImpl(KsqlConfig ksqlConfig) {
     return results;
   }
 
+  private File flushPropertiesToTempFile() throws IOException {
+    File configFile = File.createTempFile("ksqlclient", "properties");
```
Comment (big-andy-coates) [inline ksql-engine/src/main/java/io/confluent/ksql/util/KafkaConsumerGroupClientImpl.java]: Given this file contains credentials we need to be careful about the permissions on the created tmp file.

Ideally, we should be using encryption, but if we'll be switching to the Java API soon anyway, and assuming that does not require the tmp file, then it's probably sufficient for now to use  `java.nio.file.Files.createTempFile()` instead, which allows permissions to be restricted to stop other users reading the file.

File: ksql-engine/src/main/java/io/confluent/ksql/util/KafkaConsumerGroupClientImpl.java [hunk 4]
```diff
@@ -67,6 +88,20 @@ public KafkaConsumerGroupClientImpl(KsqlConfig ksqlConfig) {
...
+    OutputStreamWriter writer = new FileWriter(configFile);
```
Comment (big-andy-coates) [inline ksql-engine/src/main/java/io/confluent/ksql/util/KafkaConsumerGroupClientImpl.java]: I'd avoid `FileWriter` - it uses system default encoding.  How about `FileOutputStream` and `OutputStreamWriter` instead?

File: ksql-engine/src/main/java/io/confluent/ksql/util/KafkaConsumerGroupClientImpl.java (hunk 3 above)
Comment (apurvam) [reply ksql-engine/src/main/java/io/confluent/ksql/util/KafkaConsumerGroupClientImpl.java]: I updated the code to give only the owner read and write permissions on the file. As for encryption, the existing properties file already has credentials in unencrypted form. So this does not make anything less secure.

File: ksql-engine/src/main/java/io/confluent/ksql/util/KafkaConsumerGroupClientImpl.java [hunk 5]
```diff
@@ -88,16 +78,39 @@ public KafkaConsumerGroupClientImpl(KsqlConfig ksqlConfig) {
...
+    }
+  }
+
+  private File flushPropertiesToTempFile(Map<String, Object> configProps) throws IOException {
+    FileAttribute<Set<PosixFilePermission>> attributes
+        = PosixFilePermissions.asFileAttribute(new HashSet<>(
+            Arrays.asList(PosixFilePermission.OWNER_WRITE,
+                          PosixFilePermission.OWNER_READ)));
+    File configFile = Files.createTempFile("ksqlclient", "properties", attributes).toFile();
+    FileOutputStream outputStream = new FileOutputStream(configFile);
```
Comment (dguy) [inline ksql-engine/src/main/java/io/confluent/ksql/util/KafkaConsumerGroupClientImpl.java]: `try(FileOuputStream outputStream ...)`

File: ksql-engine/src/main/java/io/confluent/ksql/util/KafkaConsumerGroupClientImpl.java [hunk 6]
```diff
@@ -88,16 +78,39 @@ public KafkaConsumerGroupClientImpl(KsqlConfig ksqlConfig) {
...
     for (Map.Entry<String, Object> property
-        : ksqlConfig.getKsqlAdminClientConfigProps().entrySet()) {
+        : configProps.entrySet()) {
       clientProps.put(property.getKey(), property.getValue());
     }
-    clientProps.store(writer, "Configuration properties of KSQL AdminClient");
-    writer.close();
+    clientProps.store(outputStream, "Configuration properties of KSQL AdminClient");
+    outputStream.close();
     configFile.deleteOnExit();
```
Comment (dguy) [inline ksql-engine/src/main/java/io/confluent/ksql/util/KafkaConsumerGroupClientImpl.java]: move this up to line 106?

**Pull request merged**